### PR TITLE
ft: S3C-1675 HTTPS support for bucketd -> queue populator

### PR DESCRIPTION
**Pull request by jonathan-gramain**
Allow HTTPS transport for queue populator connecting to bucketd (with
optional client certificate)

### CODE REVIEW

**Comment by jonathan-gramain** [comment]
@bert-e wait

**Comment by jonathan-gramain** [comment]
@bert-e reset

**Comment by jonathan-gramain** [comment]
@bert-e wait

**Comment by jonathan-gramain** [comment]
@bert-e reset

**Comment by jonathan-gramain** [comment]
@bert-e reset

**Comment by jonathan-gramain** [comment]
@bert-e reset

**Comment by jonathan-gramain** [comment]
I successfully ran manually the core backbeat tests from Integration (on curated branch test/runBackbeatTestsOnRealEnv with one scality.cloud deployment replicating to another one) against the latest rebase of this PR on 7.4.

**Comment by jonathan-gramain** [comment]
@bert-e reset

File: extensions/replication/ReplicationConfigValidator.js [hunk 1]
```diff
@@ -1,11 +1,8 @@
 const fs = require('fs');
 const joi = require('joi');
-const { hostPortJoi, bootstrapListJoi, adminCredsJoi } =
+const { hostPortJoi, transportJoi, bootstrapListJoi, adminCredsJoi } =
```
Comment (bennettbuchanan) [inline extensions/replication/ReplicationConfigValidator.js:3]: Nice refactor/improvement along the way 👍

**Review by bennettbuchanan — APPROVED** [review]
LGTM :shipit:

**Review by philipyoo — APPROVED** [review]

**Pull request merged**